### ISSUE TITLE
ObjectPublishing PageLayouts files path problem

### ISSUE THREAD

**Issue by mrpullen** [SharePoint/PnP-Sites-Core#912]
PnP Sites Core - https://github.com/OfficeDev/PnP-Sites-Core
PnP PowerShell - https://github.com/SharePoint/PnP-PowerShell

#### Category
[X] Bug
[ ] Enhancement

#### Environment
[X] Office 365 / SharePoint Online
[?] SharePoint 2016
[X] SharePoint 2013

If SharePoint on-premises, what's exact CU version:  September CU

#### Expected or Desired Behavior
When I use Get-PnPProvisioningTemplate -Handler Publishing when I have custom Page Layouts, then attempt to Apply-PnpProvisioningTemplate I get the following error.

```
Apply-PnPProvisioningTemplate : Value cannot be null.
Parameter name: stream
At line:1 char:1
+ Apply-PnPProvisioningTemplate -Path template.xml
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Apply-PnPProvisioningTemplate], ArgumentNullException
    + FullyQualifiedErrorId : EXCEPTION,SharePointPnP.PowerShell.Commands.Provisioning.ApplyProvisioningTemplate
```

Seems to be an issue where the file folder is tokenized to {masterpagecatalog} token, then it cannot be resolved on the file system. When the file is extracted it is saved to $PWD\_catalogs\masterpage\blankWebPartPageCopy.aspx. If I copy that file to $PWD\blankWebPartPageCopy.aspx and run Apply-PnpProvisioningTEmplate it seems to run.

I am guessing that Folder isn't being reverse tokenized, or that the Folder is being improperly tokenized.

Here is the relevant template.xml

```
 <pnp:Files>
        <pnp:File Src="BlankWebPartPageCopy.aspx" Folder="{masterpagecatalog}" Overwrite="true">
          <pnp:Properties>
            <pnp:Property Key="ContentTypeId" Value="{contenttypeid:Page Layout}" />
            <pnp:Property Key="MasterPageDescription" Value="Just a custom page layout. Copied from Default Blank Webpart page layout" />
            <pnp:Property Key="Title" Value="My Custom Page Layout" />
            <pnp:Property Key="PublishingAssociatedContentType" Value=";#Welcome Page;#0x010100C568DB52D9D0A14D9B2FDCC96666E9F2007948130EC3DB064584E219954237AF390064DEA0F50FC8C147B0B6EA0636C4A7D4;#" />
            <pnp:Property Key="BSN" Value="Microsoft.SharePoint.Client.FieldLookupValue" />
            <pnp:Property Key="_ListSchemaVersion" Value="59" />
            <pnp:Property Key="_Dirty" Value="0" />
            <pnp:Property Key="_Parsable" Value="1" />
          </pnp:Properties>
        </pnp:File>
      </pnp:Files>
```

#### Steps to Reproduce

Copy the BlankWebPartPage.aspx page layout to BlankWebPartPageCopy.aspx 
Using Pnp Powershell project run the following

Get-PnpProvisioningTemplate -Out template.xml -Handlers Publishing
Apply-PnpProvisioningTemplate -Path template.xml

**Comment by mrpullen** [comment]
Debugging this, looks like the specific issue is that when the {masterpagecatalog} is parsed it is returning a relative site path (in my case /sites/test/_catalogs/masterpage) as the folder path. But the file is saved out to /_catalogs/masterpage directory instead. 

Not sure the best way to solve the issue; seems like the desired functionality, but in this case isn't quite working.

**Comment by mrpullen** [comment]
my workaround for the moment

Get-PnpProvisioningTemplate -Out template.xml -Handlers Publishing
$catalogPath = Join-Path $PWD "_catalogs/masterpage"

then use

Get-ChildItem -Path $catalogPath -Recurse | Copy-Item -Destination $PWD
or
Apply-PnPProvisioningTemplate -Path template.xml -ResourceFolder $catalogPath

**Comment by erwinvanhunen** [comment]
Hi! Thank you for reporting this. We fixed this with PR #947 and PR #933 . I will close this issue, but please do not hesitate to reopen it in case you feel the issue still stands.
